File: src/main/java/com/lee/redis/service/UserService.java
package com.lee.redis.service;

import com.lee.redis.domain.User;
import com.lee.redis.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    public List<User> findAll() {
        return userRepository.findAll();
    }

    public User findById(String id) {
        return userRepository.findById(id);
    }

    public User save(User user) {
        userRepository.save(user);
        return userRepository.findById(user.getId());
    }

    public void delete(User user) {
        userRepository.delete(user);
    }
}

File: src/main/java/com/lee/redis/service/UserService.kt
package com.lee.redis.service

import com.lee.redis.domain.User
import com.lee.redis.repository.UserRedisRepository
import org.springframework.stereotype.Service

@Service
class UserService(private val userRedisRepository: UserRedisRepository) {

    fun findAll(): MutableList<User> {
        return userRedisRepository.findAllBy()
    }

    fun findById(id: String): User {
        return userRedisRepository.findBy(id)
    }

    fun save(user: User): User {
        userRedisRepository.save(user)
        return findById(user.id)
    }

    fun delete(user: User) {
        userRedisRepository.delete(user)
    }
}
